ipatests/test_integration/test_sssd.py

```diff
@@ -233,3 +234,44 @@ def test_external_group_paging(self):
             # reset to original limit
             tasks.ldapmodify_dm(master, ldap_query.format(limit=orig_limit))
             sssd_conf_backup.restore()
+
+    def test_ext_grp_with_ldap(self):
+        """User and group with same name should not break reading AD user data.
+
+        Regression test for https://pagure.io/SSSD/sssd/issue/4073
+
+        When aduser is added in extrnal group and this is group is added
+        in group with same name of nonprivate ipa user and possix id, then
+        lookup of aduser and group should be successful when cache is empty.
+        """
+        cmd = self.master.run_command(['sssd', '--version'])
+        sssd_version = platform_tasks.parse_ipa_version(
+            cmd.stdout_text.strip())
+        if sssd_version.version <= '2.2.2':
+            pytest.skip("Fix for sssd issue:4073 unavailable in 2.2.2")
+        client = self.clients[0]
+        user = 'ipatest'
+        id = '100996'
```

Comment:
Please try to avoid shadowing builtin objects (`id` is a builtin finction). It makes code confusing.